### ISSUE TITLE
API response types should return the type requested

### ISSUE THREAD

**Issue by erik5388** [FlexSearch/FlexSearch#18]
If my title wasn't clear enough, view the photo below.

![2015-03-30_0-31-17](https://cloud.githubusercontent.com/assets/183147/6890199/6266575e-d674-11e4-86ea-8594df13e8bf.png)

Notice the section labeled "Complete Request Headers". These are the headers I pass up. I'm stating that I **accept** "application/json" responses and that I'm sending a request with an "application/json" **content-type**.

In other words... 

The response for JSON requests should be "application/json". This will be true for all cases except where the API supports additional content-types in the future.

**Comment by erik5388** [comment]
-v FlexSearch 0.21.1.0

**Comment by seemantr** [comment]
@erik5388 Thanks for pointing this out. This is a bug and I will fix it in next few days. Obviously the engine should not override the user's passed accept header unless the header is wrong.
